src/actor/mod.rs

```diff
@@ -0,0 +1,419 @@
+            builder = builder.created_at(
+                created_at
+                    .parse::<DateTime<Utc>>()
+                    .map_err(|e| anyhow::anyhow!(e.to_string()))?,
+            );
+        }
+
+        builder.initialise().map(ResponseType::CreatedAccount)
+    }
+
+    fn get_account(&self, account_id: &AccountIdentifier) -> Result<ResponseType> {
+        let account = self.account_manager.get_account(*account_id)?;
+        Ok(ResponseType::ReadAccount(account))
+    }
+
+    fn get_accounts(&self) -> Result<ResponseType> {
+        let accounts = self.account_manager.get_accounts()?;
+        Ok(ResponseType::ReadAccounts(accounts))
+    }
+
+    fn set_stronghold_password(&mut self, password: &str) -> Result<ResponseType> {
+        self.account_manager.set_stronghold_password(password)?;
+        Ok(ResponseType::StrongholdPasswordSet)
+    }
+
+    async fn send_transfer(
+        &self,
+        account_id: &AccountIdentifier,
+        transfer: &Transfer,
+    ) -> Result<ResponseType> {
+        let mut account = self.account_manager.get_account(*account_id)?;
+        let synced = account.sync().execute().await?;
+        let message = synced.transfer(transfer.clone()).await?;
+        Ok(ResponseType::SentTransfer(message))
+    }
+
+    async fn internal_transfer(
+        &self,
+        from_account_id: &AccountIdentifier,
+        to_account_id: &AccountIdentifier,
+        amount: u64,
+    ) -> Result<ResponseType> {
+        let message = self
+            .account_manager
+            .internal_transfer(*from_account_id, *to_account_id, amount)
+            .await?;
+        Ok(ResponseType::SentTransfer(message))
+    }
+}
+
+/// The wallet actor builder.
+#[derive(Default)]
+pub struct WalletBuilder {
+    rx: Option<UnboundedReceiver<Message>>,
+    message_handler: Option<WalletMessageHandler>,
+}
+
+impl WalletBuilder {
+    /// Creates a new wallet actor builder.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the receiver for messages.
+    pub fn rx(mut self, rx: UnboundedReceiver<Message>) -> Self {
+        self.rx.replace(rx);
+        self
+    }
+
+    /// Sets the wallet message handler
+    pub fn message_handler(mut self, message_handler: WalletMessageHandler) -> Self {
+        self.message_handler.replace(message_handler);
+        self
+    }
+
+    /// Builds the Wallet actor.
+    pub fn build(self) -> Wallet {
+        Wallet {
+            rx: self.rx.expect("rx is required"),
+            message_handler: WalletMessageHandler::new()
+                .expect("failed to initialise account manager"),
+        }
+    }
+}
+
+/// The Account actor.
+pub struct Wallet {
+    rx: UnboundedReceiver<Message>,
+    message_handler: WalletMessageHandler,
+}
+
+impl Wallet {
+    /// Runs the actor.
+    pub async fn run(mut self) {
+        println!("running wallet actor");
+
+        while let Some(message) = self.rx.recv().await {
```

Comment:
It might be good to receive a shutdown signal here to shutdown the `Wallet`, or it might also be good to provide an example/test case to drop the `UnboundedSender` channel, so the user can follow it easily and shutdown the `Wallet` correctly.

Reply:
moved this to an example/testcase since it's opinionated and users might use their own actor structure to use the WalletMessageHandler (like firefly uses riker's actor struct instead of this Wallet one).